### PR TITLE
sensor: hts221: Fix trigger mode and enhance testing in CI

### PR DESCRIPTION
**Pull request by erwango**
Do required changes to get this mode working
Ensure it is tested in CI

### CODE REVIEW

File: samples/sensor/hts221/prj.conf [hunk 1]
```diff
@@ -1,5 +1,7 @@
 CONFIG_STDOUT_CONSOLE=y
 CONFIG_I2C=y
 CONFIG_SENSOR=y
-CONFIG_HTS221_TRIGGER_NONE=y
+#Default trigger value:
+#CONFIG_HTS221_TRIGGER_NONE=y
+
```
Comment (MaureenHelm) [inline samples/sensor/hts221/prj.conf:6]: Why are you adding commented out code? This should be removed.

File: samples/sensor/hts221/sample.yaml [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+            - "Observation:(.*)"
+            - "Temperature:(.*) C"
+            - "Relative Humidity:(.*)%"
+tests:
+  sample.sensor.hts221:
+    extra_configs:
+        - CONFIG_HTS221_TRIGGER_NONE=y
+  sample.sensor.hts221.trigger:
+    integration_platforms:
+        - disco_l475_iot1
```
Comment (MaureenHelm) [inline samples/sensor/hts221/sample.yaml:21]: Move this to common so both tests have it as an integration platform?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: samples/sensor/hts221/sample.yaml (hunk 2 above)
Comment (erwango) [reply samples/sensor/hts221/sample.yaml]: Right, done

File: samples/sensor/hts221/prj.conf (hunk 1 above)
Comment (erwango) [reply samples/sensor/hts221/prj.conf]: Sure

**Review by MaureenHelm — DISMISSED** [review]

**Review by avisconti — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@erwango can you rebase this one?

**Review by fabiobaltieri — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**